src/Shared/FrameworkLocationHelper.cs

```diff
@@ -308,28 +308,28 @@ internal static class FrameworkLocationHelper
         /// The rules are maintained in a 2-dimensions array. Each row defines a rule. The first column
         /// defines the trigger condition. The second column defines the fallback .net and VS versions.
         /// </remarks>
-        private static readonly Tuple<Version, Version>[,] s_explicitFallbackRulesForPathToDotNetFrameworkSdkTools =
+        private static readonly (Version, Version)[,] s_explicitFallbackRulesForPathToDotNetFrameworkSdkTools =
```

Comment:
This only seems to be used in one for loop in this class. It would be messier, but do you think it would be reasonable to move this there so we don't always allocate it? I imagine that whole code path isn't used too often. At minimum, it could be cleaned up by replacing Item1, Item2 with better names.

Reply:
We can clean up ad infinitum, so one step at a time ;) I deliberately refrained from further cleanup, to make this conceptually easier to review. One semantic change per commit, one refactoring pass per PR ;)